### PR TITLE
Change type of Extensions

### LINKED ISSUE #761: Change the type of extension from Mapping to MutableMapping.
### Discussed in https://github.com/encode/httpcore/discussions/760

<div type='discussions-op-text'>

<sup>Originally posted by **karosis88** July 28, 2023</sup>
Refs: https://github.com/encode/httpx/discussions/2793</div>

---

I believe the only change needed is in the `_models.py` file.

### PR DESCRIPTION
**Pull request by karosis88**
Closes #761 

# Checklist

- [x] Change type of the extensions.
- [x] Update changelog

### CODE REVIEW

**Review by Pliner — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 
+- Change the type of `Extensions` from `Mapping[Str, Any]` to `MutableMapping[Str, Any]`.
```
Comment (Kludex) [inline CHANGELOG.md:9]:
```suggestion
- Change the type of `Extensions` from `Mapping[Str, Any]` to `MutableMapping[Str, Any]`. (#762)
```

**Review by Kludex — CHANGES_REQUESTED** [review]

**Review by Kludex — APPROVED** [review]

**Review by Kludex — APPROVED** [review]

**Comment by karosis88** [comment]
Thank you @Kludex !

**Comment by tomchristie** [comment]
Just coming back to this ...and... any reason for use to not use `Dict[...]` here instead?
Would that be preferable?

**Comment by karosis88** [comment]
It makes no difference for httpx; for httpcore, it's just a bit of reusability.

Other libraries built on top of httpcore can use more than just a dictionary.

**Pull request merged**